packages-content-model/roosterjs-content-model-editor/lib/publicApi/link/insertLink.ts

```diff
@@ -43,14 +43,16 @@ export default function insertLink(
     let url = (checkXss(link) || '').trim();
     if (url) {
         const linkData = matchLink(url);
-        const link: ContentModelLink = {
-            dataset: {},
-            format: {
-                href: linkData ? linkData.normalizedUrl : applyLinkPrefix(url),
-                anchorTitle,
-                target,
-                underline: true,
-            },
+        const link = (underline: boolean = true): ContentModelLink => {
```

Comment:
suggest move this out to be a normal function, and pass anything need as parameter.